src/main/java/com/refinedmods/refinedstorage/item/blockitem/ColoredBlockItem.java

```diff
@@ -0,0 +1,35 @@
+package com.refinedmods.refinedstorage.item.blockitem;
+
+import com.refinedmods.refinedstorage.block.BaseBlock;
+import net.minecraft.block.Block;
+import net.minecraft.item.DyeColor;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.IFormattableTextComponent;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.fml.RegistryObject;
+
+
+public class ColoredBlockItem extends BaseBlockItem {
+    private final DyeColor color;
+    private final RegistryObject<? extends Block> blockForTranslation;
+
+    public ColoredBlockItem(BaseBlock block, Properties builder, DyeColor color, RegistryObject<? extends Block> blockForTranslation) {
+        super(block, builder);
+        this.color = color;
+        this.blockForTranslation = blockForTranslation;
+    }
+
+    @Override
+    public ITextComponent getDisplayName(ItemStack stack) {
+        IFormattableTextComponent text = new StringTextComponent("");
```

Comment:
This can be calculated in the constructor. Also, try doing it this way:

```java
IFormattableTextComponent displayName = blockForTranslation.get().getTranslationKey();
if (color != DyeColor.LIGHT_BLUE) {
  displayName = new TranslationTextComponent("color.minecraft." + color.getTranslationKey()).appendString(" ").append(displayName);
}
```

Same for `ControllerBlockItem`.